crates/ra_ide_api/src/diagnostics.rs

```diff
@@ -46,6 +46,41 @@ pub(crate) fn diagnostics(db: &RootDatabase, file_id: FileId) -> Vec<Diagnostic>
             severity: Severity::Error,
             fix: Some(fix),
         })
+    })
+    .on::<hir::diagnostics::FillStructFields, _>(|d| {
+        let file_id = d.file().original_file(db);
+        let source_file = db.parse(file_id);
+        let mut buff = String::new();
+        buff.push_str("{\n");
+        for fill in &d.fills {
+            if let Some(node_ptr) = fill.value {
+                let text = node_ptr.to_node(&source_file).text();
+                if node_ptr.kind() != SyntaxKind::NAME_REF {
+                    buff.push_str(fill.name.as_str());
+                    buff.push_str(": ");
+                }
+                text.push_to(&mut buff);
+            } else {
+                buff.push_str(fill.name.as_str());
+                buff.push_str(": ()");
+            }
+            buff.push_str(",\n")
+        }
+        buff.push_str("}");
+        let syntax_node = d.syntax_node_ptr();
+        if let Some(indent) = ra_fmt::leading_indent(syntax_node.to_node(&source_file)) {
+            buff = ra_fmt::reindent(&buff, indent)
+        }
+        let mut builder = TextEditBuilder::default();
+        builder.replace(syntax_node.range(), buff);
```

Comment:
This is a bit coarse-grained: we shoudn't replace the whole struct literal, we should just append new fields before `}`

Reply:
I've made this for purpose to fill missed fields in order defined in struct even if some of them presented already.   e.g.
``` rust
struct Str {
   one: i32,
   two i32,
}
let s = Str {two: 2};

will become

let s = Str {one: (), two: 2}
but not
let s = Str { two: 2, one: ()} 
```
I do not know is it a big deal. I will remove it if it is unnecessary.

Reply:
Yeah, that's tricky... In general, it's better to make the smallest change possible to the original source-code, so as to preserve all the formatting, etc, exactly. 

Unfortunately, we don't currently have a convenient API to insert fields one by one (for the reference, here's how adding struct fields works in intellij: https://github.com/intellij-rust/intellij-rust/blob/85351520461a3cc07f7bddab8903c35d06cadf35/src/main/kotlin/org/rust/lang/core/psi/RsDefaultValueBuilder.kt#L96-L111)

I think ti makes sense, for this PR, to start small and to just add fields at the end. 

Designing a better editing API should be done separatelly, and we shouldn't try to use clever editing until we have good API for it